src/instana/instrumentation/pyramid_inst.py

```diff
@@ -0,0 +1,147 @@
+# (c) Copyright IBM Corp. 2021
+# (c) Copyright Instana Inc. 2020
+
+try:
+    from pyramid.httpexceptions import HTTPException
+    from pyramid.path import caller_package
+    from pyramid.settings import aslist
+    from pyramid.tweens import EXCVIEW
+    from pyramid.config import Configurator
+    from typing import TYPE_CHECKING, Dict, Any, Callable, Tuple
+    import wrapt
+
+    from opentelemetry.semconv.trace import SpanAttributes
+    from opentelemetry.trace import SpanKind
+
+    from instana.log import logger
+    from instana.singletons import tracer, agent
+    from instana.util.secrets import strip_secrets_from_query
+    from instana.propagators.format import Format
+
+    if TYPE_CHECKING:
+        from pyramid.request import Request
+        from pyramid.response import Response
+        from instana.span.span import InstanaSpan
+        from pyramid.registry import Registry
+
+    class InstanaTweenFactory(object):
+        """A factory that provides Instana instrumentation tween for Pyramid apps"""
+
+        def __init__(
+            self, handler: Callable[["Request"], "Response"], registry: "Registry"
+        ) -> None:
+            self.handler = handler
+
+        def _extract_custom_headers(
+            self, span: "InstanaSpan", headers: Dict[str, Any]
+        ) -> None:
+            if not agent.options.extra_http_headers:
+                return
+            try:
+                for custom_header in agent.options.extra_http_headers:
+                    if custom_header in headers:
+                        span.set_attribute(
+                            "http.header.%s" % custom_header, headers[custom_header]
+                        )
+
+            except Exception:
+                logger.debug("extract_custom_headers: ", exc_info=True)
+
+        def __call__(self, request: "Request") -> "Response":
+            ctx = tracer.extract(Format.HTTP_HEADERS, dict(request.headers))
+
+            with tracer.start_as_current_span("http", span_context=ctx) as span:
+                span.set_attribute("span.kind", SpanKind.SERVER)
+                span.set_attribute("http.host", request.host)
```

Comment:
```suggestion
                span.set_attribute(SpanAttributes.SERVER_ADDRESS, request.host)
```

Reply:
Atleast not at the moment because we set `span.data["http"]["host"]` with this [here](https://github.com/instana/python-sensor/blob/otel_migration/src/instana/span/registered_span.py#L310).